filebeat/beater/filebeat.go

```diff
@@ -329,7 +329,15 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 				return nil
 			}
 
-			stateStore.notifier.Notify(outCfg.Config())
+			// Create a new config with the output configuration. Since r.Config is a pointer, a copy is required to
+			// avoid concurrent map read and write.
+			// See https://github.com/elastic/beats/issues/42815
+			configCopy, err := conf.NewConfigFrom(outCfg.Config())
```

Comment:
a `(c *C) Clone() *C` would be quite useful here, I'm not sure what others think about it.

Reply:
Needs changes in elastic-agent-libs, which is outside the scope of this change. We can leave it as is.